typecheck/typecheck.go

```diff
@@ -133,6 +133,31 @@ func (c *Checker) stmt(s stmt.Stmt, retType *tipe.Tuple) tipe.Type {
 		}
 		if len(s.Right) == 1 && len(s.Left) == 2 && len(s.Left) == len(partials)+1 {
 			switch e := s.Right[0].(type) {
+			case *expr.Index:
+				// v, ok = m[key]
+				switch el := e.Left.(type) {
+				case *expr.Ident:
```

Comment:
I don't think you want this outer Ident check. Consider:

```
func f() map[string]int { return map[string]int{ "one": 1 } }
v, ok := f()["one"]
```

You should be able to directly look at `c.types[e.Left]` and get a `*tipe.Map` out of that.

Reply:
done.